metsim/disaggregate.py

```diff
@@ -7,107 +7,179 @@
     
+    # Interpolate the values
     try:
         interp = scipy.interpolate.PchipInterpolator(time, temp, extrapolate=True)
-        temps = interp(float(params['time_step']) * np.arange(0, len(df_disagg.index)))
+        temps = interp(ts * np.arange(0, len(df_disagg.index)))
     except ValueError:
         temps = np.full(len(df_disagg.index), np.nan)
 
     return temps
 
 
-def precip(precip):
+def precip(precip, ts):
     """
     Splits the daily precipitation evenly throughout the day
     """
-    scale = int(params['time_step']) / (consts['MIN_PER_HOUR'] * consts['HOURS_PER_DAY'])
-    return (precip*scale).resample(params['time_step']+'T').fillna(method='ffill')
+    scale = int(ts) / (cnst.MIN_PER_HOUR * cnst.HOURS_PER_DAY)
+    return (precip*scale).resample(str(int(ts))+'T').fillna(method='ffill')
 
 
-def longwave(longwave):
+def wind(wind, ts):
     """
-    Splits the daily longwave evenly throughout the day
+    Wind is assumed constant throughout the day
     """
-    return longwave.resample(params['time_step']+'T').fillna(method='ffill')
+    return wind.resample(str(int(ts))+'T').fillna(method='ffill')
 
 
-def wind(wind):
+def relative_humidity(vapor_pressure, temp):
     """
-    Wind is assumed constant throughout the day
+    TODO
     """
-    return wind.resample(params['time_step']+'T').fillna(method='ffill')
-
+    return 100. * np.minimum(vapor_pressure*1000/svp(temp), 1.0)
+   
 
-def shortwave(sw_rad, daylength, day_of_year, tiny_rad_fract):
+def vapor_pressure(hum_daily, temp, t_Tmin, n_out, ts):
+    """Calculate vapor pressure"""
+    # Scale down to milibar
+    vp_daily = hum_daily / 1000
+    # Linearly interpoolate the values
+    try:
+        interp = scipy.interpolate.interp1d(t_Tmin, vp_daily, fill_value='extrapolate')
+        vp_disagg = interp(ts * np.arange(0, n_out))
+    except ValueError:
+        vp_disagg = np.full(n_out, np.nan)
+    # Account for situations where vapor pressure is higher than saturation point
+    vp_sat = np.array(svp(temp) / 1000)
+    vp_disagg = np.where(vp_sat < vp_disagg, vp_sat, vp_disagg)
+    return vp_disagg
+
+
+
+def longwave(air_temp, vapor_pressure, tskc):
+    """ Calculate longwave """
+    emissivity_calc = {
+        'DEFAULT'    : lambda x : x,
+        'TVA'        : lambda x : 0.74 + 0.0049 * x,
+        'ANDERSON'   : lambda x : 0.68 + 0.036 * np.power(x, 0.5),
+        'BRUTSAERT'  : lambda x : 1.24 * np.power(x/air_temp, 0.14285714),
+        'SATTERLUND' : lambda x : 1.08 * (
+            1 - np.exp(-1 * np.power(x, (air_temp/2016)))),
+        'IDSO'       : lambda x : 0.7 + 5.95e-5 * x * np.exp(1500/air_temp),
+        'PRATA'      : lambda x : (1 - (1 + (46.5*x/air_temp)) *
+            np.exp(-np.sqrt((1.2 + 3. * (46.5*x/air_temp)))))
+        }
+    cloud_calc = {
+        'DEFAULT' : lambda x : (1.0 + (0.17 * tskc**2)) * x,
+        'CLOUD_DEARDORFF' : lambda x : tskc + (1-tskc)*x
+        }
```

Comment:
On 2: I just ran the checks and there's no discernible difference in speed for ~500 runs.  I'm not sure about memory, but if it becomes an issue in the future we can change it.